src/internal/utils/dom.ts

```diff
@@ -10,11 +10,17 @@ export function isContainingBlock(element: HTMLElement): boolean {
   return (
     (!!computedStyle.transform && computedStyle.transform !== 'none') ||
     (!!computedStyle.perspective && computedStyle.perspective !== 'none') ||
-    (!!computedStyle.containerType && computedStyle.containerType !== 'normal') ||
+    ![undefined, 'normal', 'inline-size'].includes(computedStyle.containerType) ||
+    // In Safari, container-type: "inline-size" creates a containing block
+    (isSafari() && ![undefined, 'normal'].includes(computedStyle.containerType)) ||
```

Comment:
What if this behavior changes in future versions? Are we going to get paged and make an emergency fix?

Reply:
Yes, this can happen. I will contribute screenshot tests for the new test page so that we will see if the behaviour changes. There is also a workaround that can be applied on the consumer side: using the `contain` with value other than "none" forces a containing block and makes behaviour consistent across browsers.

Reply:
I am drafting the alternative approach here: https://github.com/cloudscape-design/components/pull/3387